core/sequencer/trillian_client.go

```diff
@@ -141,3 +141,34 @@ func (c *MapClient) GetAndVerifyMapRootByRevision(ctx context.Context,
 	}
 	return rawMapRoot, mapRoot, nil
 }
+
+// GetMapLeavesByRevisionNoProof verifies and returns the requested map leaves at a specific revision.
+// indexes may not contain duplicates.
+func (c *MapClient) GetMapLeavesByRevisionNoProof(ctx context.Context, revision int64, indexes [][]byte) ([]*tpb.MapLeaf, error) {
+	if err := hasDuplicates(indexes); err != nil {
+		return nil, err
+	}
+	getResp, err := c.Conn.GetLeavesByRevisionNoProof(ctx, &tpb.GetMapLeavesByRevisionRequest{
+		MapId:    c.MapID,
+		Index:    indexes,
+		Revision: revision,
+	})
+	if err != nil {
+		s := status.Convert(err)
+		return nil, status.Errorf(s.Code(), "map.GetLeaves(): %v", s.Message())
```

Comment:
`GetLeavesByRevision[NoProof]`?

Also, maybe don't prefix errors returned from this method with "map.GetLeaves...". The client already knows which method they are calling, so they could add this prefix on their own, and once:

```golang
if err := c.GetMapLeavesByRevisionNoProof(...) {
	return fmt.Errorf("GetMapLeavesByRevisionNoProof: %v", err)
}
```